src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -635,7 +635,7 @@ BSONObj fillQueryServicePath(const std::vector<std::string>& servicePath)
   // Note that by construction servicePath vector must have at least one element. Note that the
   // case in which first element is "" is special, it means that the SP were not provided and
   // we have to apply the default
-  if (servicePath[0] == "")
+  if ((servicePath.size() == 0) || (servicePath[0] == ""))
```

Comment:
You shouldn't need that `(servicePath.size() == 0)`... The statement above ("Note that by construction servicePath vector must have at least one element") should be correct... and if itsn't I'd like to know in which place we are constructing a servicePath breaking this rule.